### PR TITLE
Fix SRTCP decrypt for AEAD_AES_GCM

### PR DESCRIPTION
**Pull request by at-wat**
- Fix AEAD auth tag handling
- Add test cases for AEAD_AES_GCM

See https://tools.ietf.org/html/rfc7714#section-16.1.2 about the auth tag location in AEAD_AES_GCM mode.

### CODE REVIEW

**Comment by Sean-Der** [comment]
@at-wat Could we add a conditional to the test for `isAEAD`? We could even keep it private.

This is just adding more test, or is more code coming? thanks!

File: srtcp_test.go [hunk 1]
```diff
@@ -63,182 +187,228 @@ func TestRTCPLifecycle(t *testing.T) {
 	for name, option := range options {
 		option := option
 		t.Run(name, func(t *testing.T) {
+			for caseName, testCase := range rtcpTestCases() {
+				testCase := testCase
+				t.Run(caseName, func(t *testing.T) {
+					if testCase.algo == ProtectionProfileAeadAes128Gcm {
+						t.Skip("FIXME: DecryptRTCP(nil, input, nil) for ProtectionProfileAeadAes128Gcm changes input data")
```
Comment (at-wat) [inline srtcp_test.go:194]: It should be fixed but out of the scope of this PR.

**Comment by at-wat** [comment]
> Could we add a conditional to the test for `isAEAD`? We could even keep it private.

@Sean-Der Do you mean to add `isAEAD() bool` to `ProtectionProfile`? I'm not sure how it will be used.

**Comment by Sean-Der** [comment]
I don't understand the need for `aeadAuthTagLen` vs `authTagLen`. Is there a case where both would be non-zero?

In the test why not do

```
                if cipherSuiteIsAEAD > 0 {
                    authTagPos := len(rtcpPacket) - authTagLen - srtcpIndexSize
                    copy(rtcpPacket[authTagPos:authTagPos+authTagLen], make([]byte, authTagLen))
                } else {
                    copy(rtcpPacket[len(rtcpPacket)-authTagLen:], make([]byte, authTagLen))
                }
```

**Comment by Sean-Der** [comment]
But I would love a fix for this! The more tests the better as well. If you are able to submit a fix that would be great. I can take this tomorrow also now that we have tests.

**Comment by at-wat** [comment]
RFC specifies auth tag and AEAD auth tag separately:
https://tools.ietf.org/html/rfc7714#section-14.2

> ```
>     SRTP_AEAD_AES_128_GCM
>          cipher:                 AES_128_GCM
>          cipher_key_length:      128 bits
>          cipher_salt_length:     96 bits
>          aead_auth_tag_length:   16 octets
>          auth_function:          NULL
>          auth_key_length:        N/A
>          auth_tag_length:        N/A
>          maximum lifetime:       at most 2^31 SRTCP packets and
>                                    at most 2^48 SRTP packets
> ```

as their location in the binary stream is different.

The confusion of `authTagLen` and `aeadAuthTagLen` caused a bug at
https://github.com/pion/srtp/blob/369037163aea2834a9f8d69e44d7032b1a0f6cee/srtcp.go#L12-L20

**Review by Sean-Der — APPROVED** [review]

**Pull request merged**